src/main/kotlin/app/Analytics.kt

```diff
@@ -0,0 +1,95 @@
+package app
+
+import com.github.kittinunf.fuel.core.FuelManager
+import com.github.kittinunf.fuel.core.Method
+import com.github.kittinunf.fuel.core.Request
+
+typealias Param = Pair<String, String>
+
+/**
+ * Google Analytics events tracking.
+ */
+object Analytics {
+    private val IS_ENABLED = BuildConfig.IS_GA_ENABLED
+    private val BASE_PATH = BuildConfig.GA_BASE_URL
+    private val PROTOCOL_VERSION = "1"
+    private val TRACKING_ID = BuildConfig.GA_TRACKING_ID
+    private val DATA_SOURCE = "app"
+
+    private val HIT_PAGEVIEW = "pageview"
+    private val HIT_EXCEPTION = "exception"
+
+    private val fuelManager = FuelManager()
+
+    var uuid: String = ""  // Should be set on start of the app.
+    var username: String = ""  // Should be set on success authorization.
+
+    init {
+        fuelManager.basePath = BASE_PATH
+    }
+
+    private fun post(params: List<Param>): Request {
+        return fuelManager.request(Method.POST, "/collect", params)
+    }
+
+    private fun trackEvent(event: String, params: List<Param> = listOf()) {
```

Comment:
Can you add a comment that explains how exactly this is being logged? E.g. if it's virtual urls, can you explain how it will map to virtual urls? I also suggest all virtual urls start with /virtual prefix as a convention.

Reply:
Ok.